backend/monitor_historical_writer.go

```diff
@@ -27,7 +24,95 @@ func NewMonitorHistoricalWriter(db *sql.DB) *MonitorHistoricalWriter {
 	return &MonitorHistoricalWriter{db: db}
 }
 
-func (w *MonitorHistoricalWriter) Write(historical MonitorHistorical) error {
-	// TODO: Work on me
+func (w *MonitorHistoricalWriter) Write(ctx context.Context, historical MonitorHistorical) error {
+	// Validate the historical data
+	valid, err := historical.Validate()
+	if err != nil {
+		return err
+	}
+	if !valid {
+		return nil
+	}
+
+	// Insert the historical data into the database
+	conn, err := w.db.Conn(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get connection: %w", err)
+	}
+	defer func() {
+		err := conn.Close()
+		if err != nil {
+			log.Warn().Err(err).Msg("failed to close connection")
+		}
+	}()
+
+	_, err = conn.ExecContext(ctx, "INSERT INTO monitor_historical (monitor_id, status, latency, timestamp) VALUES (?, ?, ?, ?)",
```

Comment:
Does this also need to be an UPSERT query?

Reply:
No, this is the raw data. Duplicates on here is okay.